src/core/public/keyboard_shortcut/key_parser.ts

```diff
@@ -0,0 +1,555 @@
+
+/**
+ * Keys that can be displayed with special symbols or formatting.
+ * Includes both modifier keys and special keys that have platform-specific display representations.
+ * Used to generate user-friendly display strings (e.g., '⌘' for 'cmd' on Mac).
+ */
+type DisplayMappingKeys =
+  | 'ctrl'
+  | 'alt'
+  | 'shift'
+  | 'cmd'
+  | 'enter'
+  | 'backspace'
+  | 'delete'
+  | 'tab'
+  | 'escape'
+  | 'space'
+  | 'up'
+  | 'down'
+  | 'left'
+  | 'right'
+  | 'plus';
+
+/**
+ * Platform-specific display representations for keys.
+ * Maps key names to their visual representation for each platform.
+ *
+ * @example
+ * // Mac: 'cmd' → '⌘', 'shift' → '⇧', 'up' → '↑'
+ * // Windows/Linux: 'cmd' → 'Ctrl', 'shift' → 'Shift', 'up' → '↑'
+ */
+interface PlatformDisplayMappings {
+  readonly mac: Record<DisplayMappingKeys, string>;
+  readonly other: Record<DisplayMappingKeys, string>;
+}
+
+export class KeyStringParser {
+  /**
+   * Maps KeyboardEvent.code values to normalized key names.
+   */
+  private static readonly CODE_TO_KEY_MAPPING: Record<string, string> = {
+    // Letters
+    KeyA: 'a',
+    KeyB: 'b',
+    KeyC: 'c',
+    KeyD: 'd',
+    KeyE: 'e',
+    KeyF: 'f',
+    KeyG: 'g',
+    KeyH: 'h',
+    KeyI: 'i',
+    KeyJ: 'j',
+    KeyK: 'k',
+    KeyL: 'l',
+    KeyM: 'm',
+    KeyN: 'n',
+    KeyO: 'o',
+    KeyP: 'p',
+    KeyQ: 'q',
+    KeyR: 'r',
+    KeyS: 's',
+    KeyT: 't',
+    KeyU: 'u',
+    KeyV: 'v',
+    KeyW: 'w',
+    KeyX: 'x',
+    KeyY: 'y',
+    KeyZ: 'z',
+
+    // Numbers
+    Digit0: '0',
+    Digit1: '1',
+    Digit2: '2',
+    Digit3: '3',
+    Digit4: '4',
+    Digit5: '5',
+    Digit6: '6',
+    Digit7: '7',
+    Digit8: '8',
+    Digit9: '9',
+
+    // Punctuation
+    Comma: ',',
+    Period: '.',
+    Slash: '/',
+    Semicolon: ';',
+    Quote: "'",
+    BracketLeft: '[',
+    BracketRight: ']',
+    Backslash: '\\',
+    Backquote: '`',
+    Equal: '=',
+    Minus: '-',
+
+    // Arrow keys
+    ArrowUp: 'up',
+    ArrowDown: 'down',
+    ArrowLeft: 'left',
+    ArrowRight: 'right',
+
+    // Special keys
+    Space: 'space',
+    Tab: 'tab',
+    Enter: 'enter',
+    Escape: 'escape',
+    Backspace: 'backspace',
+    Delete: 'delete',
+  };
+
+  private static readonly DISPLAY_MAPPINGS: PlatformDisplayMappings = {
+    mac: {
+      ctrl: '⌃',
+      alt: '⌥',
+      shift: '⇧',
+      cmd: '⌘',
+      enter: '↵',
+      backspace: '⌫',
+      delete: '⌦',
+      tab: '⇥',
+      escape: 'Esc',
+      space: 'Space',
+      up: '↑',
+      down: '↓',
+      left: '←',
+      right: '→',
+      plus: '+',
+    },
+    other: {
+      ctrl: 'Ctrl',
+      alt: 'Alt',
+      shift: 'Shift',
+      cmd: 'Ctrl',
+      enter: '↵',
+      backspace: '⌫',
+      delete: '⌦',
+      tab: '⇥',
+      escape: 'Esc',
+      space: 'Space',
+      up: '↑',
+      down: '↓',
+      left: '←',
+      right: '→',
+      plus: '+',
+    },
+  } as const;
```

Comment:
I think `CODE_TO_KEY_MAPPING` and `DISPLAY_MAPPINGS` can be move to `./constants.ts` to keep this file concise

Reply:
btw, `constants.ts` doesn't need an associated unit test file